language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/components/LedgerViewFlowable.java

```diff
@@ -509,4 +509,22 @@ public int hashCode() {
             return Objects.hash(commandIdToContractIds, contractIdsToCommandIds, pendingContractSet, activeContractSet);
         }
     }
+
+    /**
+     * A ledger view for unit testing of bots.
+     *
+     * @param <R> The type of the contracts in this application.
+     */
+    public static class LedgerTestView<R> extends LedgerView {
+        public LedgerTestView(PMap<String, PMap<Identifier, PSet<String>>> commandIdToContractIds,
+                   PMap<Identifier, PMap<String, PSet<String>>> contractIdsToCommandIds,
+                   PMap<Identifier, PMap<String, R>> pendingContractSet,
+                   PMap<Identifier, PMap<String, R>> activeContractSet) {
+            super(commandIdToContractIds, contractIdsToCommandIds, pendingContractSet, activeContractSet);
+        }
+
+        public LedgerTestView addActiveContractToTestView(Identifier templateId, String contractId, java.lang.Object object) {
+            return (LedgerTestView)super.addActiveContract(templateId, contractId, object);
```

Comment:
This cast will fail, because the `super` implementation always creates new `LedgerView` objects, and not `LedgerTestView`.

Reply:
Yeah. I think I need to create a wrapper class instead.

Reply:
No a wrapper, but since `LedgerView#addActiveContract` returns an instance of `LedgerView`, we need to transfer the data to `LedgerTestView`. So I guess it's both a wrapper and a subclass ;)